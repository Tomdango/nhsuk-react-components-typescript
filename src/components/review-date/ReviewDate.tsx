import React, { HTMLProps } from 'react';
import classNames from 'classnames';

interface ReviewDateProps extends HTMLProps<HTMLDivElement> {
  lastReview?: string;
  nextReview?: string;
}

const ReviewDate: React.FC<ReviewDateProps> = ({ className, lastReview, nextReview, ...rest }) => (
  <div className={classNames('nhsuk-review-date', className)} {...rest}>
    <p className="nhsuk-body-s">
      {lastReview ? `Page last reviewed: ${lastReview}` : null}
      {lastReview && nextReview ? <br /> : null}
      {nextReview ? `Next review due: ${nextReview}` : null}
    </p>
  </div>
);

export default ReviewDate;
