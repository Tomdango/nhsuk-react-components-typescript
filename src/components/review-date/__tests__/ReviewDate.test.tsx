import React from 'react';
import ReviewDate from '..';
import { shallow } from 'enzyme';

describe('ReviewDate', () => {
  it('matches snapshot', () => {
    const element = shallow(<ReviewDate />);
    expect(element).toMatchSnapshot('ReviewDate');
    element.unmount();
  });

  it('works with lastReview and nextReview', () => {
    const element = shallow(
      <ReviewDate lastReview="19 November 2019" nextReview="19 November 2020"></ReviewDate>,
    );
    expect(element.containsMatchingElement(<br />)).toBeTruthy();
    expect(element.text()).toBe(
      'Page last reviewed: 19 November 2019Next review due: 19 November 2020',
    );
    element.unmount();
  });
});
