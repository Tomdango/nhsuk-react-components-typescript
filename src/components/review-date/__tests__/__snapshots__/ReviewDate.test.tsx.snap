// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewDate matches snapshot: ReviewDate 1`] = `
<div
  className="nhsuk-review-date"
>
  <p
    className="nhsuk-body-s"
  />
</div>
`;
